docs/swagger.yaml

```diff
@@ -575,65 +757,70 @@ definitions:
+          \ once user has approved in the Vipps application."
+        example: "https://api.vipps.no/api/v1/agreements/agr_5kSeqzFAMkfBbc"
+      agreementId:
+        type: "string"
+        description: "Id of a Agreement which user may agree to. Initially the Agreement\
+          \ is in a pendingUserApproval state, and it enters active state once user\
+          \ has approved in the Vipps application."
+        example: "agr_5kSeqzFAMkfBbc"
+      redirectToVippsUrl:
+        type: "string"
+        description: "Customer should be redirected to Vipps using this URL."
+        example: "https://api.vipps.no/api/v1/register/U6JUjQXq8HQmmV"
+  Campaign:
+    type: "object"
+    properties:
+      campaignPrice:
+        type: "integer"
+        format: "int32"
+        description: "The price of the agreement in the discount period. The lowering\
+          \ of the price will be displayed in-app."
+        example: 1500
+      end:
+        type: "string"
+        format: "date-time"
+        description: "The date and time the campaign ends. Needs to be UTC"
+        example: "2019-06-01T00:00:00Z"
+  InitialCharge:
     type: "object"
     required:
     - "amount"
-    - "amountRefunded"
-    - "dueDate"
-    - "id"
-    - "status"
+    - "currency"
+    - "description"
     properties:
       amount:
         type: "integer"
         format: "int32"
         description: "Treated as a whole sum postfixed with two decimals for cents,\
           \ eg 234 = 2.34 NOK"
         example: 234
-      amountRefunded:
-        type: "integer"
-        format: "int32"
-        description: "The total amount which has been refunded, in case of status\
-          \ refund/partial refund.\nTreated as a whole sum postfixed with two decimals\
-          \ for cents, eg 234 = 2.34 NOK"
-        example: 0
-      description:
-        type: "string"
-        example: "Bla bla bla"
-      dueDate:
-        type: "string"
-        example: "2030-12-31T00:00:00Z"
-      id:
-        type: "string"
-        description: "Unique identifier for this charge, up to 15 characters."
-        example: "chg_WCVbcAbRCmu2zk"
-      status:
+      currency:
         type: "string"
+        description: "ISO-4217"
+        default: "EUR"
         enum:
-        - "PENDING"
-        - "DUE"
-        - "CHARGED"
-        - "FAILED"
-        - "CANCELLED"
-        - "PARTIALLY_REFUNDED"
-        - "REFUNDED"
-        - "PROCESSING"
-        example: "PENDING"
-      transactionId:
-        type: "string"
-        description: "Contains null until the status has reached CHARGED"
-        example: "5001419121"
-      type:
+        - "NOK"
+        pattern: "^[A-Z]{3}$"
+        example: "NOK"
+      description:
         type: "string"
-        enum:
-        - "INITIAL"
-        - "RECURRING"
-        example: "RECURRING"
-  CreateCharge:
+        description: "This field is visible to the end user in-app"
```

Comment:
Max length 45. this field is also concatinated with name

Reply:
Needs explicit description of how concatenation works. Which fields, no whitespace added, etc.